projects/blenderbot2/agents/README.md

```diff
@@ -0,0 +1,118 @@
+# BlenderBot 2 Agent Code: FAQ
+
+Below, we have compiled a list of FAQs regarding the usage of BlenderBot2. Please open an issue if your issue is not addressed below.
+
+### `ModuleNotFoundError: No module named 'transformers'`
+
+Please run `pip install transformer==4.3.3`.
+
+### `ModuleNotFoundError: No module named 'parlai.zoo.bart.bart_large`
+Please make sure you have [installed fairseq](https://github.com/pytorch/fairseq#requirements-and-installation)
+
+### `ModuleNotFoundError: No module named 'parlai.zoo.blenderbot2'`
+If you have installed ParlAI from source, make sure you have pulled from the main branch. If you have installed via pip, make sure you are on version 1.4.0 or later.
+
+### `ValueError: Must provide a valid server for search`
+You'll need to setup your own search server; see discussion in [#3816](https://github.com/facebookresearch/ParlAI/issues/3816)
+
+### `AssertionError`
+```
+    assert search_queries
+AssertionError
+```
+
+Consult "How can I use BlenderBot2 with **only** search" below.
+
+### `IndexError`
+```
+    File "/home/ParlAI/projects/blenderbot2/agents/modules.py", line 556, in <dictcomp>
+    batch_id: memory_vec[batch_id, : num_memories[mem_id]]
+IndexError: index 3 is out of bounds for dimension 0 with size 3
+```
+Make sure you're providing memories appropriately to the model; the field in which these are emitted is specified by the `--memory-key`. If you want to extract memories from the full context, set `--memory-key full_text`
+
+```
+    File "/home/ParlAI/projects/blenderbot2/agents/modules.py", line 556, in <dictcomp>
+    for batch_id, mem_id in enumerate(indices)
+IndexError: index 1 is out of bounds for dimension 0 with size 1
+```
+If you notice this during training, please set `--memory-decoder-model-file ''`.
+
+### How can I use BlenderBot2 with **only** search?
+
+You'll need to do two things:
+
+1. Set `--knowledge-access-method search_only`
+2. Set `--query-generator-model-file zoo:sea/bart_sq_gen/model`
+
+### How can I train with gold documents provided?
```

Comment:
Should we add a note here that what we mean by gold docs. Maybe to keep it general enough we could say something like this:

Gold documents are any set of documents that you need you retriever to surface. We add them to make sure the generator model sees (is conditioned on) a certain set of document. Example use cases are as follows:
1- Adding documents to the mix of retrieved that you are certain have useful knowledge that the model should use for generating response.
2- Ensure reproducibility between experiments for retrievers that have randomized responses. For example,  if you wanted your model to see the same documents that was shown to crowdsourcing agent 
 (eg `WizIntGoldDocRetrieverFiDAgent` model with `wizard of internet` dataset).

Reply:
Great idea, i'll add that note